ast.go

```diff
@@ -2337,6 +2368,24 @@ func resolveReadStatement(node *CustomFuncExpr, resolver ReadStatementResolver)
 	return "", fmt.Errorf("custom function %s is not resolvable", node.Name)
 }
 
+// resolveReadStatementParam returns a function that acts like an iterator.
+// Every time the function is called it gets the next bind value.
+func resolveReadStatementParam(resolver ReadStatementResolver) func() (string, error) {
```

Comment:
we make use of a closure to capture the state of `bindValues` and `i` and move forward with every function call